tests/base/test_array.py

```diff
@@ -0,0 +1,467 @@
+    def test_mcse_methods(self, array_stats, method):
+        rng = np.random.default_rng(42)
+        ary = rng.normal(size=(4, 100))
+        prob = 0.95 if method == "quantile" else None
+        result = array_stats.mcse(ary, method=method, prob=prob)
+        assert result.shape == ()
+        assert result > 0
+
+    def test_mcse_invalid_method(self, array_stats):
+        ary = np.random.randn(4, 100)
+        with pytest.raises(ValueError, match="must be one of"):
+            array_stats.mcse(ary, method="invalid")
+
+
+class TestParetoFunctions:
+    def test_pareto_min_ss(self, array_stats):
+        rng = np.random.default_rng(42)
+        ary = rng.normal(size=(4, 100))
+        result = array_stats.pareto_min_ss(ary)
+        assert result.shape == ()
+        assert result > 0
+
+    def test_psislw_basic(self, array_stats):
+        rng = np.random.default_rng(42)
+        ary = rng.normal(size=(100,))
+        log_weights, khat = array_stats.psislw(ary)
+        assert log_weights.shape == (100,)
+        assert khat.shape == ()
+
+    def test_psislw_r_eff(self, array_stats):
+        rng = np.random.default_rng(42)
+        ary = rng.normal(size=(100,))
+        log_weights, khat = array_stats.psislw(ary, r_eff=0.5)
+        assert log_weights.shape == (100,)
+        assert khat.shape == ()
+
+    def test_pareto_khat(self, array_stats):
+        rng = np.random.default_rng(42)
+        ary = rng.normal(size=(4, 100))
+        result = array_stats.pareto_khat(ary)
+        assert result.shape == ()
+
+
+class TestPowerScale:
+    def test_power_scale_lw(self, array_stats):
+        rng = np.random.default_rng(42)
+        ary = rng.normal(size=(100,))
+        result = array_stats.power_scale_lw(ary, alpha=0.5)
+        assert result.shape == (100,)
+
+    @pytest.mark.filterwarnings("ignore::RuntimeWarning")
+    def test_power_scale_sense(self, array_stats):
+        rng = np.random.default_rng(42)
+        ary = rng.normal(size=(4, 100))
+        lower_w = rng.normal(size=(4, 100))
+        upper_w = rng.normal(size=(4, 100))
+        result = array_stats.power_scale_sense(ary, lower_w, upper_w, -0.5, 0.5)
+        assert result.shape == ()
+
+
+class TestComputeRanks:
+    def test_compute_ranks_basic(self, array_stats):
+        x = np.array([3.0, 1.0, 2.0, 4.0])
+        result = array_stats.compute_ranks(x)
+        assert result.shape == (4,)
+
+    def test_compute_ranks_relative(self, array_stats):
+        x = np.array([3.0, 1.0, 2.0, 4.0])
+        result = array_stats.compute_ranks(x, relative=True)
+        assert np.all(result >= 0)
+        assert np.all(result <= 1)
+
+    @pytest.mark.parametrize("axis", [0, 1, -1])
+    def test_compute_ranks_axis(self, array_stats, axis):
+        rng = np.random.default_rng(42)
+        ary = rng.normal(size=(10, 20, 30))
+        result = array_stats.compute_ranks(ary, axis=axis)
+        assert result.ndim == ary.ndim
```

Comment:
given the shape is always preserved independently of what is passed as `axis` I would check the maximum value within the array which will depend on the length of the dimension indicated by `axis`